### PR TITLE
[tmva] Fix tmva DL packages for new cudnn v9

### PR DESCRIPTION
**Pull request by lmoneta**
This PR migrates the TMVA Deep Learning classes (RNN) for new cud version 9

### CODE REVIEW

File: tmva/tmva/src/DNN/Architectures/Cudnn/RecurrentPropagation.cu [hunk 1]
```diff
@@ -391,22 +487,28 @@ void TCudnn<AFloat>::InitializeRecurrentDescriptors(TDescriptors *&descriptors,
 template<typename AFloat>
 void TCudnn<AFloat>::ReleaseRNNDescriptors(TDescriptors * descriptors)
 {
-   auto rnnDescriptors = static_cast<RNNDescriptors_t *>(descriptors);
-   CUDNNCHECK(cudnnDestroyRNNDescriptor(rnnDescriptors->LayerDescriptor));
+#if 0
```
Comment (guitargeek) [inline tmva/tmva/src/DNN/Architectures/Cudnn/RecurrentPropagation.cu:490]: Can you maybe add an inline comment here on why the code was "commented" out?

**Comment by guitargeek** [comment]
Spin-off PR: https://github.com/root-project/root/pull/15602

**Review by guitargeek — APPROVED** [review]
Thank you so much for working on this! I have tested the changes by running these tutorials when building ROOT with `tmva-gpu` and `cudnn`:
```txt
tutorials/tmva/TMVA_CNN_Classification.C
tutorials/tmva/TMVA_RNN_Classification.C
```

Note: this PR needs to be backported to ROOT 6.32.

One minor comment in-line, if you feel like addressing it.

Please also take a look at this related PR: https://github.com/root-project/root/pull/15602 Thanks!

**Comment by dpiparo** [comment]
Can we please merge this and its spinoff https://github.com/root-project/root/pull/15602 and backport to 6.32 ASAP?

**Comment by dpiparo** [comment]
Can this be merged and backported please? @lmoneta

**Pull request merged**